connectors/src/connectors/snowflake/temporal/cast_known_errors.ts

```diff
@@ -15,10 +15,13 @@ function isSnowflakeExpiredPasswordError(
   err: unknown
 ): err is SnowflakeExpiredPasswordError {
   return (
-    typeof err === "object" &&
-    err !== null &&
-    "code" in err &&
-    err.code === 390106 // this is the magic code number for an expired password error, the message says "Specified password has expired.  Password must be changed using the Snowflake web console."
+    err instanceof Error &&
```

Comment:
nit: can we change the `SnowflakeExpiredPasswordError` such that `name` is of type "OperationFailedError" ?